xcm/xcm-builder/src/currency_adapter.rs

```diff
@@ -33,16 +35,48 @@ enum Error {
 
 impl From<Error> for XcmError {
 	fn from(e: Error) -> Self {
+		use XcmError::FailedToTransactAsset;
 		match e {
-			Error::AssetNotFound => XcmError::FailedToTransactAsset("AssetNotFound"),
-			Error::AccountIdConversionFailed =>
-				XcmError::FailedToTransactAsset("AccountIdConversionFailed"),
-			Error::AmountToBalanceConversionFailed =>
-				XcmError::FailedToTransactAsset("AmountToBalanceConversionFailed"),
+			Error::AssetNotFound => FailedToTransactAsset("AssetNotFound"),
+			Error::AccountIdConversionFailed => FailedToTransactAsset("AccountIdConversionFailed"),
+			Error::AmountToBalanceConversionFailed => FailedToTransactAsset("AmountToBalanceConversionFailed"),
 		}
 	}
 }
 
+/// Simple adapter to use a currency as asset transactor. This type can be used as `type AssetTransactor` in
+/// `xcm::Config`.
+///
+/// # Example
+/// ```
+/// use frame_support::parameter_types;
+/// use xcm::v0::{MultiLocation, Junction};
+/// use xcm_builder::{ParentIsDefault, CurrencyAdapter, IsConcrete};
+///
+/// /// Our chain's account id.
+/// type AccountId = sp_runtime::AccountId32;
+///
+/// /// Our relay chain's location.
+/// parameter_types! {
+///     RelayChain: MultiLocation = MultiLocation::X1(Junction::Parent);
+/// }
+///
+/// /// Some items that implement `Convert<MultiLocation, AccountId>`. Can be more, but for now we just assume we accept
+/// /// messages from the parent (relay chain).
+/// pub type LocationConvertor = (ParentIsDefault<RelayChain>);
+///
+/// /// Final current adapter. This can be used in `xcm::Config` to specify how asset related transactions happen.
+/// pub type AssetTransactor = CurrencyAdapter<
+///     // Use this currency:
+///     u128,
+///     // The matcher: use the currency when the asset is a concrete asset in our relay chain.
+///     IsConcrete<RelayChain>,
+///     // The local convertor: default account of the parent relay chain.
+///     LocationConvertor,
+///     // Our chain's account ID type.
+///     AccountId,
+/// >;
```

Comment:
```suggestion
/// /// Final currency adapter. This can be used in `xcm::Config` to specify how asset related transactions happen.
/// pub type AssetTransactor = CurrencyAdapter<
///     // Use this balance type:
///     u128,
///     // The matcher: use the currency when the asset is a concrete asset in our relay chain.
///     IsConcrete<RelayChain>,
///     // The local convertor: default account of the parent relay chain.
///     LocationConvertor,
///     // Our chain's account ID type.
///     AccountId,
/// >;
```